operators/src/source/ogr_source.rs

```diff
@@ -423,6 +426,15 @@ where
         // TODO: add OGR time filter if forced
         let dataset = Self::open_gdal_dataset(&dataset_information)?;
         let mut layer = dataset.layer_by_name(&dataset_information.layer_name)?;
+        let use_ogr_spatial_filter = dataset_information.force_ogr_spatial_filter
+            || layer.has_capability(gdal::vector::LayerCaps::OLCFastSpatialFilter);
+
+        if use_ogr_spatial_filter {
+            // rectangular geometry from West, South, East and North values.
+            let filter_geometry = query_rectangle.bbox.try_into()?;
+            // TODO: log uses spatial filter
+            layer.set_spatial_filter(&filter_geometry);
```

Comment:
Why does this work? `filtere_geometry` is dropped after this if-block… so is `set_spatial_filter` copying the geometry?

Reply:
it works... maybe the compiler is smart enough. maybe it is a bug

Reply:
use https://gdal.org/doxygen/ogr__api_8h.html#a5cba569e0779a02a95327f041d9f7a13?

Reply:
Furthermore, we should add a TODO, that the OGR-filter may be inaccurately allowing more features that should be returned in a "strict" fashion, whatever this means. cf . the docu